m2cgen/assemblers/boosting.py

```diff
@@ -317,15 +317,28 @@ def _assemble_tree(self, tree):
         op = ast.CompOpType.from_str_op(tree["decision_type"])
         assert op == ast.CompOpType.LTE, "Unexpected comparison op"
 
-        # Make sure that if the "default_left" is true the left tree branch
-        # ends up in the "else" branch of the ast.IfExpr.
-        if tree["default_left"]:
-            op = ast.CompOpType.GT
-            true_child = tree["right_child"]
-            false_child = tree["left_child"]
+        missing_type = tree['missing_type']
```

Comment:
I wonder whether we can eliminate the logic duplication here by merging 2 conditions into one:
```
if missing_type in ("NaN", "None"):
    reverse_condition = missing_type == "NaN" and tree["default_left"]
    reverse_condition |= missing_type == "None" and tree["threshold"] >= 0
    if reverse_condition:
        op = ast.CompOpType.GT
        true_child = tree["right_child"]
        false_child = tree["left_child"]
    else:
        true_child = tree["left_child"]
        false_child = tree["right_child"]
else:
    raise ValueError('Unknown missing_type')
```
Please also note that we use double quotes for string literals.

Reply:
Got it. I'll update pull request with fixes and unit test in a couple of days.

Reply:
Thanks! Unit test you can place in the following file:
https://github.com/BayesWitnesses/m2cgen/blob/train_test_split/tests/assemblers/test_lightgbm.py